wallet/chainntfns.go

```diff
@@ -822,10 +822,10 @@ func (w *Wallet) VoteOnOwnedTickets(winningTicketHashes []*chainhash.Hash,
 		addrmgrNs := dbtx.ReadBucket(waddrmgrNamespaceKey)
 		for i, ticketHash := range ticketHashes {
 			ticketPurchase, err := w.TxStore.Tx(txmgrNs, ticketHash)
-			if err != nil || ticketPurchase == nil {
+			if errors.Is(errors.NotExist, err) && ticketPurchase == nil {
 				ticketPurchase, err = w.StakeMgr.TicketPurchase(dbtx, ticketHash)
 			}
-			if err != nil {
+			if !errors.Is(errors.NotExist, err) {
```

Comment:
should remain `if err != nil {`, since the error may be the result of either the Tx or TicketPurchase calls.